instrumentation/lettuce/lettuce-4.0/src/main/java/io/opentelemetry/auto/instrumentation/lettuce/v4_0/InstrumentationPoints.java

```diff
@@ -38,11 +36,8 @@
   private static final Set<CommandType> NON_INSTRUMENTING_COMMANDS = EnumSet.of(SHUTDOWN, DEBUG);
 
   public static SpanWithScope beforeCommand(final RedisCommand<?, ?, ?> command) {
-    final String spanName = command == null ? "Redis Command" : command.getType().name();
-    final Span span =
-        LettuceClientDecorator.TRACER.spanBuilder(spanName).setSpanKind(CLIENT).startSpan();
-    LettuceClientDecorator.DECORATE.afterStart(span);
-    return new SpanWithScope(span, currentContextWith(span));
+    final Span span = LettuceDatabaseClientTracer.TRACER.startSpan(null, command, null);
```

Comment:
```suggestion
    final Span span = LettuceDatabaseClientTracer.TRACER.startSpan(/* connection= */ null, command, /* originType= */ null);
```

Reply:
So many optional parameters does make this API a bit suspicious - would be good to confirm we got the right abstraction here. I sort of wonder if it's possible for the instrumentation tracer to define the API expected for the instrumentation rather than leaking the base class API over here.

Reply:
Partly you are correct and I want to remove that `originType` as soon as I understand WTH is this.

Partly Lettuce instrumentations are somewhat special in that they have connection info only in some places and not another. Otherwise I still think that `startSpan(Connection, Query)` is a good API.

Reply:
Makes sense. Still recommend the "param docs" to make this a but clearer.